### DISCUSSION TITLE
Payment gateway

### DISCUSSION THREAD

**Discussion by faibacon** [hotspotbilling/phpnuxbill#1] [Q&A]
Hi. I am so happy for your project it has made me learn about api a lot. Kindly tell me if it is possible if i would use a kenyan mobile payment gateway called **### **_m-pesa_****? The system you used for payment gateway code am not familiar with it. Kindly help. Many thanks 😊 🙏 😊 🙏

**Comment by ibnux** [comment]
it is possible
as long as there are API to request payment to user and API to check is payment paid or not.

Because most Billing installed on Private Network, so it must have manual check paid payment.

**Comment by Focuslinkstech** [comment]
> Hi. I am so happy for your project it has made me learn about api a lot. Kindly tell me if it is possible if i would use a kenyan mobile payment gateway called **### **_m-pesa_****? The system you used for payment gateway code am not familiar with it. Kindly help. Many thanks 😊 🙏 😊 🙏

What about paystack.com, it use mpesa as payment channel
Paystack.com gateway is under development
Let me know your opinion on mpesa and paystack.com

**Reply by patricknana532** [reply]
Please need help I want to add my mikrotik router to PHPNEXBILL that I already host in my hosting. how do I do it please?

**Reply by Focuslinkstech** [reply]
> Please need help I want to add my mikrotik router to PHPNEXBILL that I already host in my hosting. how do I do it please?

In other to use add your router you need to have public ip address or use VPN if you are behind the NAT

**Reply by patricknana532** [reply]
I tried to use the VPN with the DDNS that Mikrotik provides for public IP but it doesn't work. Please can you help me or guide me through the configuration using the two methods you mentioned.

**Reply by Focuslinkstech** [reply]
> I tried to use the VPN with the DDNS that Mikrotik provides for public IP but it doesn't work. Please can you help me or guide me through the configuration using the two methods you mentioned. 

First method
contact your Internet Provider tell them you need public ip address.

Second method 
1: use third party VPN service provider 
Like hostddns.us, remotewinbox.com etc

2: host your own private VPN server

**Reply by patricknana532** [reply]
Thank you very much, I will try and get back to you

**Comment by islamifty** [comment]
Can you pls add something like manual payment ....

**Reply by ibnux** [reply]
That voucher code you can sell to your customers

**Comment by winter9176** [comment]
kak request menu donk untuk paket pelanggan vpn biar bisa auto nat di firewall berdasarkan pool vpn

**Comment by winter9176** [comment]
satu lagi kak request multi payment gateway biar bisa pakay gateway lebih dari 1 payment contoh duitku & paypall bisa aktif bersamaan di menu pembayaran, harapan semoga bisa jadi referensi buat kakak, sukses buat phpnuxbilling

**Comment by Focuslinkstech** [comment]
> Hi. I am so happy for your project it has made me learn about api a lot. Kindly tell me if it is possible if i would use a kenyan mobile payment gateway called **### **_m-pesa_****? The system you used for payment gateway code am not familiar with it. Kindly help. Many thanks 😊 🙏 😊 🙏

Mpesa payment gateway is now available, but you have purchased it

**Reply by zadok68** [reply]
I am interested in this module for mpesa please guid me on how to make the payment

**Reply by zadok68** [reply]
you can reach me out on my email 
makorizadok282@gmail.com

**Comment by metromatika** [comment]
hi @ibnux , saya menggunakan payment gateway duitku, akan tetapi setelah create transaksi dia gagal, sepertinya ada kesalahan di sistem
untuk akses xendit bekerja normal. apakah kamu bisa memperbaiki iini??

**Comment by juliusjob324** [comment]
any help.....payment gateways to pay wifi via kcb or mpesa

**Reply by Focuslinkstech** [reply]
Yes, its available and its paid plugin

**Reply by juliusjob324** [reply]
which plugin is it

**Reply by Focuslinkstech** [reply]
Check plugin manager from nuxbill dashboard

**Reply by SAID6716** [reply]
Hi @Focuslinkstech Please Dont Forget to Create another payment System For Tanzania Payment Gateway which Include Zenopay, Azampay, Selcome, Clickpesa PesaPal ans So many More We are In tanzania We Want to Use Your Module But We Fail to Use Of Lack Of Payment Gateway That Can Support it

**Reply by Focuslinkstech** [reply]
You can contact for personalized support, we can integrate the payment system for you

**Comment by SAID6716** [comment]
Hi @ibnux and @Focuslinkstech  Am From Tanzania and am Really Interested On Using This System Billing but am New In this Situation but facing A Techinical Challenges on How i can Connect to My Home Small Router Mikrotik and phpnux bill with Login Captive Portal and i dont have a Public IP Static from my ISP
Second is How and Which is  Best Hosting  to Host This Billing Thanks

**Reply by Focuslinkstech** [reply]
you can just come to the Telegram Community [https://t.me/phpnuxbill](https://t.me/phpnuxbill), Members can assist you on issues that you are facing

**Reply by mdahadhossen** [reply]
Piprapay plugin phpnuxbill installed how

**Comment by djpao29** [comment]
🚀 Feature Request: Automatic Simple Queue Assignment on Client Creation (PPPoE / VPN / Hotspot)
Description

Currently, phpNuxBill allows client provisioning using different access methods such as PPPoE, VPN, and Hotspot, along with service plan assignment.
However, there is no native option to automatically create and assign a MikroTik Simple Queue when a client is added.

This feature request proposes adding an option to automatically manage Simple Queues during client provisioning.

Proposed Functionality

When creating a client via PPPoE, VPN, or Hotspot, phpNuxBill should allow:

Automatic creation of a Simple Queue on the associated MikroTik router.

Selection of bandwidth control type, including:

PCQ (Per Connection Queue)

Max-Limit (fixed upload/download limits)

Configuration of common Simple Queue parameters:

target (client IP or address-list)

max-limit

limit-at (optional)

queue-type (PCQ or default)

priority (optional)

comment (client identifier)

Benefits

Enables use cases where bandwidth control is based on Simple Queues instead of PPP Profiles.

Provides greater flexibility for ISP network designs.

Improves automation and consistency during client provisioning.

Reduces manual configuration and operational errors on MikroTik routers.

Useful for mixed environments (PPPoE, Hotspot, VPN).

Implementation Considerations

The option could be configured:

At the service plan level, or

Directly in the client creation form.

Must support MikroTik RouterOS v6 and v7.

For PPPoE, the feature should work independently of PPP Profiles, allowing both methods to coexist.

For Hotspot users, the Simple Queue could be bound by:

Static IP

DHCP lease

Address-list

Include validation to:

Prevent duplicate Simple Queues

Sync queue state on suspend, enable, or delete client

Additional Suggestion

Optional synchronization between phpNuxBill client status and MikroTik Simple Queue (disable/enable/remove automatically).

**Comment by SAID6716** [comment]
it may work in Tanzania Mobiles Operators
